src/components/forms/formComponents.js

```diff
@@ -0,0 +1,387 @@
+import { h, div, label, input, span, button } from 'react-hyperscript-helpers';
+
+import { cloneDeep, isNil, isEmpty, isString } from 'lodash/fp';
+import Creatable from 'react-select/creatable';
+import Select from 'react-select';
+import { FormValidators } from './forms';
+import {RadioButton} from '../RadioButton';
+
+import './formComponents.css';
+
+const styles = {
+  inputStyle: {
+    padding: '25px 15px',
+    width: '100%'
+  }
+};
+
+//---------------------------------------------
+// Form Behavior
+//---------------------------------------------
+const validateFormInput = (config, value) => {
+  const { setError, validators } = config;
+  if (validators) {
+    const validationResults = validators
+      .filter(validator => !validator.isValid(value))
+      .map(x => x.msg);
+
+    const isValid = validationResults.length === 0;
+    setError(isValid ? undefined : validationResults);
+    return isValid;
+  }
+  setError();
+  return true;
+};
+
+const normalizeValue = (value) => {
+  if (typeof value === 'string') {
+    return value.trim();
+  } else if (Array.isArray(value)) {
+    return value
+      .map(x => normalizeValue(x))
+      .filter(x => x); // filter out null strings
+  }
+  return value;
+};
+
+const onFormInputChange = (config, value) => {
+  const { id, onChange, setFormValue } = config;
+  const normalizedValue = normalizeValue(value);
+  const isValidInput = validateFormInput(config, normalizedValue);
+
+  onChange({key: id, value: normalizedValue, isValid: isValidInput });
+  setFormValue(value);
+};
+
+const errorMessage = (error) => {
+  return error && div({ className: `error-message fadein`}, [
+    span({ className: 'glyphicon glyphicon-play' }),
+    ...error.map((err) => div([err])),
+  ]);
+};
+
+//---------------------------------------------
+// Form Controls
+//---------------------------------------------
+export const formInputGeneric = (config) => {
+  const {
+    id, title, type, disabled,
+    placeholder,
+    inputStyle, ariaDescribedby,
+    formValue, error, setError
+  } = config;
```

Comment:
Is there no way to enforce values for config? `id` for example is required, but `type` and `placeholder` is not. Also, there is a one-of condition for `placeholder` and `title`. Perhaps some default values might be necessary? I'm not opposed to this as is, but I do think there will be hard-to-debug bugs when devs forget to add all the right configs for the specific form entry they're using.

Reply:
possibly, we could have this information in the FormField type declaration: 
```js
{
  ...,
  requiredProps: ['id'],
  customPropValidation: (props) => { 
    if (!isNil(props.title) && !isNil(props.placeholder)) {
      throw 'both title and placeholder cannot be specified';
    }
  }
}
```

thoughts?

Reply:
Something like that should work, but maybe that's a little more than this PR is intended to do? Might be a good follow-on ticket.

Reply:
That's a really great suggestion, Conor! Would like to add that in the next update :D